### DISCUSSION TITLE
AppProject as a self-service for developers

### DISCUSSION THREAD

**Discussion by reggie-k** [argoproj/argo-cd#8031] [Ideas]
With RBAC for ArgoCD applications, and latest changes with repositories and clusters that developers can configure for themselves, being independent of ArgoCD admins, and having a self-service approach to AppProject creation and management seems beneficial.

Something similar to ApplicationSet for AppProjects (AppProjectSet?) combined with RBAC might achieve this goal.
As an admin, I would like to define what values can be filled in a template for AppProject (similar to the AppSet generators), and then define some project templates for the developers.

The developers then will be able to fill values in those templates, and will need ArgoCD RBAC to allow them managing only AppProjects from those templates (maybe such template might be a new RBAC resource and AppProject might be then a subresource of the template).

The approach will allow me, as an admin, to protect the cluster in terms of setting the whitelist of allowed cluster-scoped resources the developers can touch, and potentially define a regex of allowed target namespaces for the AppProj , while letting the developers create the AppProjects themselves and also manage themselves which other users/groups will be on the AppProject role. The developers might also fill values for their target ns name, by the regex an admin set on the template.

Kind of a freedom in a well-defined boundaries.

**Comment by crenshaw-dev** [comment]
@reggie-k, work-in-progress proposal for project inheritance / App-of-Projects restrictions: https://github.com/argoproj/argo-cd/pull/9992

**Comment by reggie-k** [comment]
Thanks a lot, will look into it thoroughly Sunday
